packages/framework-integration-tests/integration/providers/aws/subscriptions.integration.ts

```diff
@@ -63,6 +70,50 @@ describe('subscriptions', () => {
         clientB.disconnect()
       }
     })
+
+    it('should delete connection data when socket is disconnected', async () => {
+      const connectionsCount = await countConnectionsItems()
+      const client = await graphQLClientWithSubscriptions()
+      try {
+        await waitForIt(countConnectionsItems, (newCount) => newCount == connectionsCount + 1)
+        client.disconnect()
+        await waitForIt(countConnectionsItems, (newCount) => newCount == connectionsCount)
+      } catch {
+        client.disconnect()
+      }
+    })
+  })
+
+  describe('when socket reconnects ', () => {
+    let client: DisconnectableApolloClient
+    before(async () => {
+      client = await graphQLClientWithSubscriptions()
+    })
+    after(() => {
+      client.disconnect()
+    })
+
+    it('keeps the same subscriptions', async () => {
+      const cartID = random.uuid()
+      const originalSubscriptionsCount = await countSubscriptionsItems()
+      // Let's create two subscriptions to the same read model
+      const observableOne = cartSubscription(client, cartID)
+      const observableTwo = cartSubscription(client, cartID)
+      // Call the subscribe function to send the subscription to server
+      observableOne.subscribe(() => {})
+      observableTwo.subscribe(() => {})
+      // Wait for for the subscriptions to arrive
+      await waitForIt(countSubscriptionsItems, (newCount) => newCount == originalSubscriptionsCount + 2)
+      // Check we receive data when the read model is modified
+      await cartMutation(client, cartID)
+      await expect(Promise.all([promisify(observableOne), promisify(observableTwo)])).to.eventually.be.fulfilled
+
+      // Now reconnect and see if we keep the having the same subscription and receive data
+      await client.reconnect()
+      await waitForIt(countSubscriptionsItems, (newCount) => newCount == originalSubscriptionsCount + 2)
+      await cartMutation(client, cartID)
+      await expect(Promise.all([promisify(observableOne), promisify(observableTwo)])).to.eventually.be.fulfilled
```

Comment:
Might not mean much, but when I look at these tests I feel a lot of joy

Reply:
That's the joy of the elder developers that lies very deep in our hearts